packages/rule-doc-generator/src/Finder/ClassByTypeFinder.php

```diff
@@ -36,10 +42,12 @@ public function findByType(array $directories, string $type): array
                 continue;
             }
 
-            $desiredClasses[] = $class;
+            $relativeFile = str_replace($workingDirectory, '', $file);
+
+            $desiredClasses[$relativeFile] = $class;
```

Comment:
Hiding non-key values into keys might cause WTF later.
This should be a value object instead.